src/monster.h

```diff
@@ -407,7 +407,7 @@ class monster : public Creature, public JsonSerializer, public JsonDeserializer
 
     private:
         int hp;
-        std::unordered_map<std::string, mon_special_attack> special_attacks;
+        std::map<std::string, mon_special_attack> special_attacks;
```

Comment:
Forgot to explain this.

It's not directly related to the whole PR, but from what I could find, `unordered_map` needs to hash the entire key. In the case of string keys, that is roughly O(len(k)) complexity, where k is the key.
`map` only needs to read O(log n) symbols.

Since `special_attacks` is usually a very small container, it is highly likely that hashing will take more time than binary tree search.